.github/workflows/test.yml

```diff
@@ -0,0 +1,87 @@
+name: Build and run tests
+
+on: [push, pull_request]
```

Comment:
Given a number of PRs come from branches within this repo, it may be worth changing this to something like the following. This will prevent such PRs from running through tests twice.
```
...
on:
  push:
    branches:
      - master
  pull_request:
jobs:
...
```